src/main/java/org/moltimate/moltimatebackend/Structure/StructureUtils.java

```diff
@@ -0,0 +1,184 @@
+     *
+     * @param structure:   structure to search for amino acid in
+     * @param residueName: name of residue to search for
+     * @return list of residues with given name inside structure
+     */
+    public static List<Group> getResiduesByType(Structure structure, String residueName) {
+        ArrayList<Group> results = new ArrayList<>();
+        for (Chain chain : structure.getChains()) {
+            for (Group residue : chain.getAtomGroups(GroupType.AMINOACID)) {
+                if (residue.getChemComp().getThree_letter_code().equals(residueName)) {
+                    results.add(residue);
+                }
+            }
+        }
+        return results;
+    }
+
+    /**
+     * Get all atoms of a certain type inside a structure
+     *
+     * @param structure: structure to search for atoms in
+     * @param atomType:  type of atom to search for in structure
+     * @return a list of atoms in the structure matching the specified type
+     */
+    public static List<Atom> getAtomByType(Structure structure, String atomType) {
+        ArrayList<Atom> atoms = new ArrayList<>();
+        structure.getChains().forEach(chain ->
+                chain.getAtomGroups(GroupType.AMINOACID).forEach(group ->
+                        atoms.addAll(group.getAtoms().stream()
+                                .filter(atom -> atom.getName().equals(atomType))
+                                .collect(Collectors.toList()))));
+        return atoms;
+    }
+
+    /**
+     * Get all atoms of a certain type inside a residue
+     *
+     * @param residue:  residue to search for atoms in
+     * @param atomType: type of atom to search for in structure
+     * @return a list of atoms in the structure matching the specified type
+     */
+    public static List<Atom> getAtomByType(Group residue, String atomType) {
+
+        return residue.getAtoms().stream()
+                .filter(atom -> atom.getName().equals(atomType)).collect(Collectors.toList());
+    }
+
+    /**
+     * Creates a map of residue -> location where each location is the center of each residue
+     *
+     * @param residues: list of residues to create locations for
+     * @return map of residue -> double[] where each array is a 3 dimensional coordinate
+     * representing the center of that residue.
+     */
+    public static Map<Group, double[]> residueToLocationMap(List<Group> residues) {
+        HashMap<Group, double[]> locationMap = new HashMap<>();
+        residues.forEach(residue -> locationMap.put(residue, getResidueLocation(residue)));
+        return locationMap;
+    }
+
+    /**
+     * Root mean squared distance between two points
+     *
+     * @param point1: first point
+     * @param point2: second point
+     * @return root mean squared distance between the two points
+     */
+    public static double rmsd(double[] point1, double[] point2) {
+        //Point3d[] p1 = {new Point3d(point1)};
```

Comment:
Is it worthwhile to keep this commented code or should it be deleted?

Reply:
Whoops, magmacode. Got rid of it